modules/sdk-coin-algo/src/algo.ts

```diff
@@ -542,6 +606,172 @@ export class Algo extends BaseCoin {
+    const isUnsignedSweep = this.isValidPub(params.userKey) && this.isValidPub(params.backupKey);
+
+    // Validate the root address
+    if (!this.isValidAddress(params.rootAddress)) {
+      throw new Error('invalid rootAddress, got: ' + params.rootAddress);
+    }
+
+    // Validate the destination address
+    if (!this.isValidAddress(params.recoveryDestination)) {
+      throw new Error('invalid recoveryDestination, got: ' + params.recoveryDestination);
+    }
+
+    if (params.firstRound && params.lastRound) {
+      if (new BigNumber(params.firstRound).isNegative()) {
+        throw new Error('first round needs to be a positive value');
+      }
+      if (new BigNumber(params.lastRound).isNegative()) {
+        throw new Error('first round needs to be a positive value');
+      }
+
+      if (params.firstRound > params.lastRound) {
+        throw new Error('firstRound cannot be greater than or equal to lastRound');
+      }
+    }
+
+    const genesisId = params.genesisId ?? (this.bitgo.getEnv() === 'prod' ? MAINNET_GENESIS_ID : TESTNET_GENESIS_ID);
+    const genesisHash =
+      params.genesisHash ?? (this.bitgo.getEnv() === 'prod' ? MAINNET_GENESIS_HASH : TESTNET_GENESIS_HASH);
+
+    Utils.validateBase64(genesisHash);
+
+    if (!isUnsignedSweep && !params.walletPassphrase) {
+      throw new Error('walletPassphrase is required for non-bitgo recovery');
+    }
+
+    const factory = new AlgoLib.TransactionBuilderFactory(coins.get('algo'));
+    const txBuilder = factory.getTransferBuilder();
+
+    let userPrv: string | undefined;
+    let backupPrv: string | undefined;
+    if (!isUnsignedSweep) {
+      if (!params.bitgoKey) {
+        throw new Error('bitgo public key from the keyCard is required for non-bitgo recovery');
+      }
+      try {
+        userPrv = this.bitgo.decrypt({ input: params.userKey, password: params.walletPassphrase });
+        backupPrv = this.bitgo.decrypt({ input: params.backupKey, password: params.walletPassphrase });
+        const userKeyAddress = Utils.privateKeyToAlgoAddress(userPrv);
+        const backupKeyAddress = Utils.privateKeyToAlgoAddress(backupPrv);
+        txBuilder.numberOfRequiredSigners(2).setSigners([userKeyAddress, backupKeyAddress, params.bitgoKey]);
+      } catch (e) {
+        throw new Error(
+          'unable to decrypt userKey or backupKey with the walletPassphrase provided, got error: ' + e.message
+        );
+      }
+    }
+
+    // TODO: Remove the ! after making nodeParams mandatory
+    const token =
+      this.bitgo.getEnv() === 'prod'
+        ? params.nodeParams!.token
+        : '2810c2d168e8417c5f111d38d68327b8cfe2d0ddc02986490c22f8ddf4128bcd';
+    const baseServer = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.baseServer : 'http://localhost/';
+    const port = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.port : 8443;
+
+    const client = new algosdk.Algodv2(token, baseServer, port);
```

Comment:
Replace with a getClient method and validate the inputs
```suggestion
    const token =
      this.bitgo.getEnv() === 'prod'
        ? params.nodeParams!.token
        : '2810c2d168e8417c5f111d38d68327b8cfe2d0ddc02986490c22f8ddf4128bcd';
    const baseServer = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.baseServer : 'http://localhost/';
    const port = this.bitgo.getEnv() === 'prod' ? params.nodeParams?.port : 8443;

    const client = new algosdk.Algodv2(token, baseServer, port);
```

Reply:
i've removed all this code since nodeParams is now required